### ISSUE TITLE
[terra-clinical-data-grid] Remove onRequestContent scroll lazy loading

### ISSUE THREAD

**Issue by dkasper-was-taken** [cerner/terra-clinical#660]
# Feature Request
Remove `onRequestContent` pagination according to UX feedback around progressive disclosure driven by scrolling.

## Description
The scroll driven pagination should be removed, but alternate methods of dealing with large data sets should be discussed as well.

Following UX feedback inline with this discussion:
https://github.com/cerner/terra-framework/issues/1043

The onRequestContent behavior similar to infinite list should be deprecated.
Associated deprecation issue:
https://github.com/cerner/terra-framework/issues/1088

## @ Mentions
@scottwilmarth
